runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -125,6 +127,17 @@ private void handleEviction(RemovalNotification<Long, ILogData> notification) {
         }
     }
 
+    private double calculateEstimatedAvgEntrySize() {
+        if (readCache.size() == 0) {
+            return 0.0;
+        }
+        long currentCacheSize = readCache.size();
+        long currentDataSizeInCache = readCache.asMap().entrySet().stream()
+                .mapToLong(e -> sizeOf.deepSizeOf(e.getKey()) + sizeOf.deepSizeOf(e.getValue())).sum();
+
+        return (double) currentDataSizeInCache / currentCacheSize;
+    }
```

Comment:
This will be computed every minute, but the cache eviction happens at a much higher rate therefore this will be inaccurate.

Reply:
Another concern is that this will block the evicted objects from being garbage collected since sizeOf.deepSizeOf() still holds the references to them. This issue is serious when readCache itself is large and the eviction rate is high. Yet considering that we don't enable this metrics in production I think it should be fine.